File: drive-ui/src/app/model/Shareable.ts
export class Shareable {
  name: string;
  path: string;
  isFile: boolean;
  created: Number;
  lastModified: Number;
  size: Number;

  type: string;
  icon: string;

  public constructor(init?:Partial<Shareable>) {
    Object.assign(this, init);
  }
}

File: drive-server/src/main/java/com/fmning/drive/dto/Shareable.java
package com.fmning.drive.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Shareable {
    private String name;
    private String path;
    private Boolean isFile;
    private long created;
    private long lastModified;
    private long size;
}
